src/peft/utils/save_and_load.py

```diff
@@ -45,6 +45,35 @@ def get_embedding_layer_name(model, layer, is_embedding_in_target_modules):
     return None
 
 
+def get_buffers_to_save(model: torch.nn.Module, config) -> dict[str, torch.Tensor]:
+    """
+    Get the buffers to save for the given model and config.
+
+    Although the base model weights are not updated when using PEFT, some buffers may still be updated and therefore
+    need to be saved as part of the PEFT checkpoint. An example of this are the running stats of batch norm layers.
+
+    Args:
+        model (torch.nn.Module):
+            The PEFT model.
+        config (PeftConfig):
+            The PEFT config.
+
+    Returns:
+        dict[str, torch.Tensor]:
+            The buffers to save.
+    """
+    # note: config is currently not used but that may change in the future
+    buffers_to_save = {}
+    for module_name, module in model.named_modules():
+        # currently, we only deal with running stats from BatchNorm* modules
+        if not hasattr(module, "track_running_stats"):
+            continue
+        for buffer_name, buffer in module.named_buffers():
```

Comment:
do we wanna check if `track_running_stats` is set to `True`?

Reply:
Good point. I considered this, but was hesitant. Say someone trains a model with tracking enabled, and then turns it off for some reason before saving. Then we would not save these buffers even though they're needed, do I see that right? Ideally, we would have a check if they changed vis-à-vis the base model, but I don't see a way to monitor this except for storing a copy of all these buffers, requiring extra memory.

Reply:
yes... true.. ok its safer to save them you are right yeah someone can then turn it off and it starts to use the batch's summary statistics in inference mode...